backend/src/managers/UserCache.ts

```diff
@@ -12,7 +13,27 @@ export class UserCache {
         this.userRepository = userRepository;
     }
 
+    private initialize() {
+        if (this.initialized === true) {
+            return Promise.resolve();
+        }
+        if (this.initialized) {
+            return this.initialized;
+        }
+        this.initialized = (async () => {
+            const users = await this.userRepository.getLastActiveUsers();
+            for (const user of users) {
+                this.cache(this.mapUserRaw(user));
+            }
+            this.initialized = true;
+        })();
```

Comment:
it works but it feels a bit hacky. Having two properties, one for the action and another one for status would probably make the code more readable

Reply:
I think of this `initialized` property as a state machine:
`false` → (awaitable `Promise`) → `true`

I think spitting it into two separate properties introduces the possibility of having unwanted states, and the readability should be addressed in another way. I'll try giving this prop another name and adding some comments.

Reply:
Please check if now it's more clear.